src/v/security/credential_store.h

```diff
@@ -50,7 +50,7 @@ class credential_store {
     }
 
     template<typename T>
-    auto get(const credential_user& name) -> std::optional<T> const {
+    auto get(const credential_user& name) const -> std::optional<T> const {
```

Comment:
is this `const` different than the one after the trailing return type?

Reply:
Yep -- this const is the one that makes the method callable on a `const credential_store`.  I think in the trailing return type style, the last const is specifying that the returned value is const.

Reply:
Yes. Which is strange. I think it may have been misplaced initially.

Reply:
interesting!